nft_ingester/src/program_transformers/bubblegum/decompress.rs

```diff
@@ -23,17 +23,25 @@ where
         return Ok(());
     }
 
+    // Start a db transaction.
+    let multi_txn = txn.begin().await?;
+
     // Partial update of asset table with just leaf.
-    upsert_asset_with_leaf_info_for_decompression(txn, id_bytes.to_vec()).await?;
+    upsert_asset_with_leaf_info_for_decompression(&multi_txn, id_bytes.to_vec()).await?;
 
     upsert_asset_with_compression_info(
-        txn,
+        &multi_txn,
         id_bytes.to_vec(),
         false,
         false,
         1,
         Some(id_bytes.to_vec()),
         true,
     )
-    .await
+    .await?;
+
+    // Close out transaction and relinqish the lock.
+    multi_txn.commit().await?;
```

Comment:
Need to ensure that we know fully what happens if the TX fails and the reason for the TX failing. Is this a recoverable TX error or not?

Reply:
Does my explanation above for how SeaORM transactions are rolled back when the object goes out of scope answer the question?  Is there some type of recovery we should aim to do, which is different then the current code that afaik just ignores errors and continues on?